File: app/src/test/java/bj/discogsbrowser/marketplace/ListingFactory.java
package bj.discogsbrowser.marketplace;

import bj.discogsbrowser.model.listing.Listing;
import bj.discogsbrowser.model.listing.Price;
import bj.discogsbrowser.model.listing.Release;
import bj.discogsbrowser.model.listing.Seller;

/**
 * Created by Josh Laird on 09/05/2017.
 */

public class ListingFactory
{

    public Listing getListing()
    {
        Listing listing = new Listing();
        Seller seller = new Seller();
        seller.setUsername("username");
        listing.setComments("");
        listing.setSeller(seller);
        listing.setCondition("goodCondition");
        listing.setSeller(getSeller());
        listing.setUri("uri");
        listing.setPrice(getPrice());
        listing.setRelease(getRelease());
        return listing;
    }

    public Seller getSeller()
    {
        Seller seller = new Seller();
        seller.setShipping("shipping");
        seller.setUsername("username");
        return seller;
    }

    public Price getPrice()
    {
        Price price = new Price();
        price.setValue(1.00);
        return price;
    }

    public Release getRelease()
    {
        Release release = new Release();
        release.setDescription("");
        release.setThumbnail("");
        return release;
    }
}

File: app/src/testShared/bj/discogsbrowser/listing/ListingFactory.kt
package bj.discogsbrowser.listing

import bj.discogsbrowser.model.common.Price
import bj.discogsbrowser.model.common.Seller
import bj.discogsbrowser.model.listing.Listing
import bj.discogsbrowser.order.OrderFactory.buildListingRelease

/**
 * Created by Josh Laird on 19/05/2017.
 */
object ListingFactory {
    @JvmStatic fun buildNumberOfListings(number: Int): List<Listing> {
        return (1..number).map { buildListing(it.toString()) }
    }

    @JvmStatic fun buildListing(id: String): Listing {
        val listing = Listing()
        listing.id = id
        listing.condition = "aite"
        listing.sleeveCondition = "less aite"
        listing.comments = "'dis shit aite'- Joy Orbison"
        listing.uri = "http://aiteshit.com"
        listing.release = buildListingRelease("description")
        listing.seller = buildSeller()
        listing.price = Price("", 50.0)
        return listing
    }

    @JvmStatic fun buildSeller(): Seller {
        val seller = Seller()
        seller.username = "that's wassup"
        seller.shipping = "AiteExpress"
        return seller
    }
}